src/Cli/ConfigGenerator.cs

```diff
@@ -1005,76 +1016,92 @@ private static bool TryUpdateConfiguredMcpValues(
+                    updatedMcpOptions = updatedMcpOptions! with { DmlTools = DmlToolsConfig.FromBoolean((bool)updatedValue) };
+                    _logger.LogInformation("Updated RuntimeConfig with Runtime.Mcp.Dml-Tools as '{updatedValue}'", updatedValue);
+                    return true; // Return early since we're setting all tools at once
                 }
 
-                // Runtime.Mcp.Dml-Tools.Describe-Entities
+                // Handle individual tool updates
+                bool? describeEntities = currentDmlTools?.DescribeEntities;
+                bool? createRecord = currentDmlTools?.CreateRecord;
+                bool? readRecord = currentDmlTools?.ReadRecord;
+                bool? updateRecord = currentDmlTools?.UpdateRecord;
+                bool? deleteRecord = currentDmlTools?.DeleteRecord;
+                bool? executeRecord = currentDmlTools?.ExecuteRecord;
+
                 updatedValue = options?.RuntimeMcpDmlToolsDescribeEntitiesEnabled;
                 if (updatedValue != null)
                 {
-                    McpDmlToolsOptions dmlToolsOptions = updatedMcpOptions?.DmlTools ?? new();
-                    dmlToolsOptions = dmlToolsOptions with { DescribeEntities = (bool)updatedValue };
-                    updatedMcpOptions = updatedMcpOptions! with { DmlTools = dmlToolsOptions };
+                    describeEntities = (bool)updatedValue;
+                    hasToolUpdates = true;
                     _logger.LogInformation("Updated RuntimeConfig with Runtime.Mcp.Dml-Tools.Describe-Entities as '{updatedValue}'", updatedValue);
                 }
 
-                // Runtime.Mcp.Dml-Tools.Create-Record
                 updatedValue = options?.RuntimeMcpDmlToolsCreateRecordEnabled;
                 if (updatedValue != null)
                 {
-                    McpDmlToolsOptions dmlToolsOptions = updatedMcpOptions?.DmlTools ?? new();
-                    dmlToolsOptions = dmlToolsOptions with { CreateRecord = (bool)updatedValue };
-                    updatedMcpOptions = updatedMcpOptions! with { DmlTools = dmlToolsOptions };
+                    createRecord = (bool)updatedValue;
+                    hasToolUpdates = true;
                     _logger.LogInformation("Updated RuntimeConfig with Runtime.Mcp.Dml-Tools.Create-Record as '{updatedValue}'", updatedValue);
                 }
 
-                // Runtime.Mcp.Dml-Tools.Read-Record
                 updatedValue = options?.RuntimeMcpDmlToolsReadRecordEnabled;
                 if (updatedValue != null)
                 {
-                    McpDmlToolsOptions dmlToolsOptions = updatedMcpOptions?.DmlTools ?? new();
-                    dmlToolsOptions = dmlToolsOptions with { ReadRecord = (bool)updatedValue };
-                    updatedMcpOptions = updatedMcpOptions! with { DmlTools = dmlToolsOptions };
+                    readRecord = (bool)updatedValue;
```

Comment:
`readRecords` Please make the tool name change everywhere its needed. same for `execute-entity`